docs/userguide/CDN/USERGUIDE.md

```diff
@@ -0,0 +1,289 @@
+
+```php
+$services = $cdnService->listServices();
+foreach ($services as $service) {
+    /** @var $service OpenCloud\CDN\Resource\Service **/
+}
+```
+
+[ [Get the executable PHP script for this example](/samples/CDN/list-services.php) ]
+
+### Get a service
+
+You can retrieve a specific service by using that service's ID, as shown in the following example:
+
+```php
+$service = $cdnService->getService('0e09ad12-2bfe-4607-80fd-116fa68d9c79');
+/** @var $service OpenCloud\CDN\Resource\Service **/
+```
+
+[ [Get the executable PHP script for this example](/samples/CDN/get-service.php) ]
+
+### Update a service
+
+This operation takes one parameter, an associative array, with the following keys:
+
+| Name | Description | Data type | Required? | Default value | Example value |
+| ---- | ----------- | --------- | --------- | ------------- | ------------- |
+| `domains` | An array of associative arrays, each specifying a domain name for your service. | Array of associative arrays | Yes | `null` | `array( array( 'domain' => 'www.acme.com' ), array ( 'domain' => 'acme.com' ) )` |
+| `origins` | An array of associative arrays, each specifying an origin server for your service. The `port`, `ssl`, and `rules` elements for each origin server are optional. | Array of associative arrays | Yes | `null` | `array( array( 'origin' => 'origin.acme.com', 'port' => 80, 'ssl' => false, 'rules' => array() ) )` |
+| `flavorId` | The flavor used to configure this service. Use the [list flavors](#list-flavors) operation to retrieve a list of all available flavors. | String | Yes| `null` | `cdn` |
+| `restrictions` | An array of associative arrays, each specifying a restriction for who can or cannot access content from the CDN cache. | Array of associative arrays | No | `null` | `array( array( 'name' => 'website only', 'rules' => array( array( 'name' => 'mywebsite.com', 'httpHost' => 'www.mywebsite.com' ) ) ) )` |
+| `caching` | An array of associative arrays, each specifying a caching rule for your service's assets. | No | `null` | `array( array( 'name' => 'default', 'ttl' => 3600 ) )` |
+
+You can update a service as shown in the following example:
+
+```php
+$service->update(array(
+    'origins' => array(
+        array(
+            'origin' => '44.33.22.11',
+            'port'   => 80,
+            'ssl'    => false
+        )
+    )
+));
+```
+
+[ [Get the executable PHP script for this example](/samples/CDN/update-service.php) ]
+
+### Delete a service
+
+You can delete a service as shown in the following example:
+
+```php
+$service->delete();
+```
+
+[ [Get the executable PHP script for this example](/samples/CDN/delete-service.php) ]
+
+## Service Assets
+A service will have its assets distributed and cached across a CDN's edge nodes.
+
+### Purge all cached service assets
+
+You can purge all cached assets of a service as shown in the following example:
+
+```php
+$service->purgeAssets();
```

Comment:
My understanding is that requesting assets via their CDN URLs is done out-of-band from the API. One of the pieces of information this API will return is the CDN domain name when you perform a `$cdnService->getService({serviceId})` operation. After that point, the user can simply suffix the relative path of their asset to this domain name.

When all assets are purged via the `$service->purgeAssets()` operation, if the user were to request that asset via the asset URL, the service would not find the asset in its caches (located on edge nodes) and try to request it from the origin server at that point. 

This is my understanding but perhaps @amitgandhinz can confirm.